subsys/net/lib/sockets/sockets.c

```diff
@@ -285,6 +398,28 @@ ssize_t zsock_sendto(int sock, const void *buf, size_t len, int flags,
 	return len;
 }
 
+#ifdef CONFIG_USERSPACE
+Z_SYSCALL_HANDLER(zsock_sendto, sock, buf, len, flags, dest_addr, addrlen)
+{
+	struct sockaddr_storage dest_addr_copy;
+
+	if (Z_SYSCALL_OBJ(sock, K_OBJ_NET_CONTEXT)) {
+		errno = EBADF;
+		return -1;
+	}
+
+	Z_OOPS(Z_SYSCALL_MEMORY_READ(buf, len));
+	if (dest_addr) {
+		Z_OOPS(z_user_from_copy(&dest_addr_copy, (void *)dest_addr,
```

Comment:
How do we handle here (and in other places) a case when addrlen > sizeof(dest_addr_copy) ?

Reply:
if z_user_from_copy() fails the calling thread is killed.

Reply:
Oh I see it, yes this needs to be accounted for.